app/views/posts/show.html.erb

```diff
@@ -6,16 +6,30 @@
   </div> 
  
   <p class="post-content2"><%= @post.text %></p>
+   
 </ul>
+<%= button_to 'Like', like_post_path(@post), method: :put, class: "like-button" %>
   
-  <ul class="comments2">
   <h2>Comments</h2>
+  <ul class="comments">
     <% @post.comments.each do |comment| %>
       <li>
         <strong><%= comment.user.name %>:</strong>
         <p><%= comment.text %></p>
       </li>
     <% end %>
   </ul>
-  <%= link_to 'Back', request.referer || root_path %>
+
+  
+  <%= form_with(model: [@post, Comment.new]) do |form| %>
+    <div class="field">
+      <%= form.label :text, "New Comment" %>
+      <%= form.text_area :text %>
+    </div>
+    <div class="actions">
+      <%= form.submit "Add Comment" %>
+    </div>
+  <% end %>
```

Comment:
- [ ] Great progress on implementing the forms to enhance the user experience! 👍 However, for optimal user experience and to align seamlessly with the design, I'd like to suggest a few improvements:

1. **Route Placement:**
   Consider rendering your `form` at its dedicated routes. Placing it on the post-show page might clutter the layout. Leverage the routes you've defined in `config/routes` for comments. This ensures a cleaner structure and aligns with the separation of concerns. Follow the pattern you've established for the post model.

2. **View Organization:**
   Ensure the `form` is created in the right place within your views, specifically in `views/comments/new.html.erb`. This provides clarity and maintains a logical structure in your codebase.

By making these adjustments, not only will your code be more organized and maintainable, but it will also enhance the overall user experience. Happy coding! 😊